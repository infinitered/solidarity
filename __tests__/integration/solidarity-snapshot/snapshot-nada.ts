import execa from 'execa'
import tempy from 'tempy'

const SOLIDARITY = `${process.cwd()}/bin/solidarity`
const origCwd = process.cwd()
let originalTimeout

beforeAll(() => {
  // These can be slow on CI
  originalTimeout = jasmine.DEFAULT_TIMEOUT_INTERVAL
  jasmine.DEFAULT_TIMEOUT_INTERVAL = 90000
  const tempDir = tempy.directory()
  process.chdir(tempDir)
})

afterAll(function() {
  // Fix timeout change
  jasmine.DEFAULT_TIMEOUT_INTERVAL = originalTimeout
})

test('solidarity report works', async done => {
  try {
    execa.shell(`echo n | ${SOLIDARITY} snapshot`).then(result => {
      // check a few from the report
      expect(result.stdout).toMatchSnapshot()
      expect(result.code).toBe(0)
      done()
    })
  } catch (err) {
    done.fail()
  }
})

afterEach(() => {
  process.chdir(origCwd)
})
