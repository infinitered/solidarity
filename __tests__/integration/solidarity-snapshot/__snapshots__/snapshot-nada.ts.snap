// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`solidarity report works 1`] = `
"[36m?[39m [1mNo \`.solidarity\` file found for this project.  Would you like to create one?[22m [2m(
Y/n) [22m[1A[5D[86C[1B[1000D[K[1A[1000D[K[36m?[39m [1mNo \`.solidarity\` file found for this project.  Would you like to create one?[22m [2m(
Y/n) [22m[1A[5D[86C[1B[1000D[K[1A[1000D[K[36m?[39m [1mNo \`.solidarity\` file found for this project.  Would you like to create one?[22m [79D[79C[1000D[K[36m?[39m [1mNo \`.solidarity\` file found for this project.  Would you like to create one?[22m [36mf
alse[39m[1A[4D[85C
[?25hNothing to do Â¯\\\\_(ãƒ„)_/Â¯"
`;
